### ISSUE TITLE
org-roam-dalies-capture-today inserts subheading where insertion point in daily file is instead of top level heading appended to file

### ISSUE THREAD

**Issue by nickanderson** [org-roam/org-roam#1499]
### Description

Running v2 branch, `org-roam-dailies-capture-today` inserts wherever the insertion point inside that buffer is. It works as expected if the the insertion point in that buffer was left at the top of the file, but if I had navigated in the buffer for todays file elsewhere, it will insert as a subheading of that.

#### Steps to Reproduce

Use the following `org-roam-dailies-capture-templates` definition from https://org-roam.discourse.group/t/v2-error-running-org-roam-dailies-find-today/1511

```
  (setq org-roam-dailies-capture-templates
        `(("d" "default" entry
           "*  %?"
           :if-new (file+head ,(expand-file-name "%<%Y-%m-%d>.org" org-roam-dailies-directory)
                              "#+title: Work log for %<%Y-%m-%d>")
           :clock-in t
           :clock-resume t)
          ))
```

- Run `org-roam-dailies-find-today.
- Ensure there are at least 3 headings
- Move the insertion point to the second heading
- Run `org-roam-dailies-capture-today` See that the capture is not appended to the end of the file, but inserted as a subheading instead.

#### Backtrace

It does not error, just does not do what is expected.

#### Expected Results

The daily capture is found as the last top level heading.

#### Actual Results

The daily capture is found as a subheading.

### Environment

- Emacs: GNU Emacs 26.3 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.8)
 of 2019-09-16
- Framework: Spacemacs
- Org: Org mode version 9.4.5 (9.4.5-29-gc43041-elpa @ /home/nickanderson/.emacs.d/elpa/26.3/develop/org-20210426/)
- Org-roam: 2.0.0

Installed by patching packages.el
```
modified   layers/+emacs/org/packages.el
@@ -67,7 +67,9 @@
     (org-trello :toggle org-enable-trello-support)
     (org-sticky-header :toggle org-enable-sticky-header)
     (verb :toggle org-enable-verb-support)
-    (org-roam :toggle org-enable-roam-support)
+    (org-roam :toggle org-enable-roam-support
+              :location (recipe :fetcher github :repo "org-roam/org-roam" :branch "v2")
+              )
```

- Org-roam commit: https://github.com/jethrokuan/org-roam/commit/0a7d365e226b59674a97baddb8b3419762a00b6e

**Comment by AloisJanicek** [comment]
I was experiencing something similar and I noticed that something is moving the point after the capture is done. Because if the point is at the beginning of the heading line when capture is initiated, descendant heading is inserted instead of the top level one. I double-checked the `org-capture-after-finalize-hook` and realized that one function I've added there moves the point. I fixed it by using `save-excursion`. So not exactly the org-roam fault, I suspect this is natural `org-capture` behavior.
